pkg/secrethub/credentials/sessions/session_aws.go

```diff
@@ -35,13 +38,13 @@ func (s *awsSessionCreator) Create(httpClient *http.Client) (Session, error) {
 
 	getCallerIdentityReq, err := getCallerIdentityRequest(region, s.awsConfig...)
 	if err != nil {
-		return nil, err
+		return nil, handleAWSErr(err)
 	}
 
 	req := api.NewAuthRequestAWSSTS(api.SessionTypeHMAC, region, getCallerIdentityReq)
 	resp, err := httpClient.CreateSession(req)
 	if err != nil {
-		return nil, err
+		return nil, handleAWSErr(err)
```

Comment:
I'm not really sure whether we do this correctly in other places, but I think it's best if we use this function directly after calling AWS functionality. That way we avoid placing them at many different places, just to be sure we catch all errors. In this case that would mean placing this function at line 67 and 76 because those are the only calls to the `aws` package.

Reply:
Good point :+1: I also think that that's the best approach.